### ISSUE TITLE
Ignore changes?

### ISSUE THREAD

**Issue by kris4de** [versionpress/versionpress#1185]
I have a shortcode on a page that inputs content of other pages into this one. Think of it as global content so I don't have to edit it on every page.

Interesting is, every time a user visits a page with this shortcode, VersionPress tracks changes of every page the shortcode is copying.

**Example:**

There are three shortcodes on a page. Every shortcode inserts content of another page (H2, Button and Featured).

Now a user visits the page with the shortcodes on it and VersionPress commits following change on each copied page: `post_modified` and `post_modified_gmt`.

![versionpress-commit](https://cloud.githubusercontent.com/assets/5444765/22013734/d72f8a64-dc9a-11e6-9a89-23cd233ea8c2.png)

But in WordPress, these pages don't have the same "post modified" dates.

![pages](https://cloud.githubusercontent.com/assets/5444765/22013733/d72b2564-dc9a-11e6-9869-fd738768e440.png)

I use the latest WordPress version with latest Git installed. Theme is called X with page builder Cornerstone (works flawless, awesome!).

This would mess up the history of VersionPress with every page visit and I think it would cost too much server resources. Any way to ignore those changes?

**Comment by pavelevap** [comment]
Hi,

there is no `post_modified` date displayed on second screenshot, because in WordPress administration you can see only `post_date` (date of publishing particular post/page).

So, VersionPress tracks these changes correctly, because it is possible that theme/plugin (source of shortcode) is calling `wp_update_post()` or any other function which updates post data even if there is no change (only `post_modified` is changed in the end). It is probably not needed and should be optimized by theme authors.

I noticed the same behaviour when updating post (click "Update" button) without any change. I would suggest to ignore these changes by VersionPress in the future, there is no need to track them.

**Comment by kris4de** [comment]
Yes, my fault – good morning from Germany. I thought a bit longer about it. The global blocks are set to private to prevent page views and the shortcode sets them to public to access the contents. I have a newer version that don't need to update the post. I will test that ...

But the ability to ignore only post_modified changes would be nice. As you said, WordPress writes this date even without other changes made.

**Comment by pavelevap** [comment]
Great! It is not the best aproach to update post/page during every request. VersionPress only shows that something is wrong in this case, but changes are probably tracked correctly.

I like the idea of ignoring `post_modified` changes when there is no other data change. I am not sure about all possible consequences, but I will try to look at it.

**Comment by borekb** [comment]
Maybe [column ignoring](https://github.com/versionpress/versionpress/blob/325cd8335b4083a17d2e489c5b72b018f7f103eb/docs/Plugin-Support.md#ignoring-columns) would help here?

**Comment by kris4de** [comment]
I tested my new plugin and nothing is updated every time, now.

Column ignoring could work. But ideally, the developers of those plugins should create the `yml` files, right?

Little side note: In another WordPress installation, I noticed that the commit message of edited posts always says `Edited option rs_tables_created`. And not `Edited post POSTNAME`. Same versions installed.

**Comment by borekb** [comment]
Hm, those have the same priority of 12 ([posts](https://github.com/versionpress/versionpress/blob/325cd8335b4083a17d2e489c5b72b018f7f103eb/plugins/versionpress/.versionpress/actions.yml#L14), [options](https://github.com/versionpress/versionpress/blob/325cd8335b4083a17d2e489c5b72b018f7f103eb/plugins/versionpress/.versionpress/actions.yml#L32)), if they happen at the same time the order is "random". Is some plugin changing both the option and the post at the same time?

**Comment by kris4de** [comment]
This happens with Cornerstone and TinyMCE, but the theme – X – is changing some category thing.

![option](https://cloud.githubusercontent.com/assets/5444765/22018267/4ec5bbd0-dcae-11e6-878f-a3b8dd016e16.png)

Perhaps change the priority to better identify commits? `Edited Post XY` sounds better than `Edited option AB`, I think.

**Comment by pavelevap** [comment]
There should be both messages, "Edited post" and "Edited option" when you switch from "Full diff" view to "Overview"? Commit title is probably random due to the same priority of both actions.

But, as I can see on the screenshot, there is nothing changed for `rs_tables_created.ini` option? I am not sure how it is caused, because when there is no change of option value, message should not be displayed. One possible explanation is that this option is (probably incorrectly) changed several times during one request (with no change in the end)?

**Comment by kris4de** [comment]
Yes, open the tab and all commit messages are displayed. But for better overview, I think another commit message should appear as tab title. As a user, I search for posts, not options. Priority to "Edited post" could help to find the right commit.

Interesting. Do you mean that changed options should appear in another file?

**Comment by borekb** [comment]
That depends on the context. Imagine a plugin where the main action is changing an option and also updating some post is secondary. So ideally, in the future, the plugin that does this should set the priorities correctly.
